### ISSUE TITLE
Deferred loading of RTL text plugin causes flicker for raster layers

### ISSUE THREAD

**Issue by ChrisLoer** [maplibre/maplibre-gl-js#2380]
When a deferred load of the RTL text plugin completes, there's logic that causes the source caches to reload:

https://github.com/maplibre/maplibre-gl-js/blob/9f713cd2299a567e4f2a8000d984cd3f4ca14a19/src/style/style.ts#L226-L238

The purpose of this logic is to cause them to regenerate their symbol buckets with the new RTL shaping applied. If the plugin loads before the initial tile load, this will be a no-op, but if some tiles have already loaded it will reparse them. This is some extra work and the tile cache is cleared out, but the old tiles can still display until the new tiles are ready.

This logic is unnecessary for raster sources, since they don't have any symbols to regenerate. Also, raster tiles tend to load faster (because no parsing/bucket-generation is necessary), and are thus more likely to load before the RTL plugin loads. This generates extra work, but might not be that noticeable except for the fact that reloading a raster tile also resets the tile fade animation to the beginning.

The upshot of all this is an unsightly flicker during map load (in this example I'm just loading a single satellite layer and setting the rtl text plugin).

![load_flicker](https://user-images.githubusercontent.com/375121/231302433-9bf59f27-17b6-4fbd-9924-15e0141f9194.gif)

Stepping through in the debugger, you can see loaded tiles getting dropped one at a time as reloaded images arrive:

![reloading_raster_tiles](https://user-images.githubusercontent.com/375121/231302788-6c19505f-796c-4014-9002-2c3981d112bf.gif)

And this is what the load looks like with the `reload()` call turned off for raster sources (PR coming soon):

![load_no_flicker](https://user-images.githubusercontent.com/375121/231303296-86619e83-4cef-4c57-9586-86a10da07d55.gif)

**Comment by HarelM** [comment]
So basically turn this off on case of raster is the solution here?
Feel free to open a PR.
I can assign a bounty to it if needed.

**Comment by HarelM** [comment]
Sorry, saw the PR only after I wrote my comment... It's still valid though.

**Comment by ChrisLoer** [comment]
>Sorry, saw the PR only after I wrote my comment... It's still valid though.

Is the comment about the code pen for repro or bounty? It's a tiny fix.

Here's a codepen: https://codepen.io/ChrisLoer/pen/KKGdYjv 

I suggest turning on network throttling to make it easier to see, e.g. here's with "Fast 3g"

![codepen_throttled](https://user-images.githubusercontent.com/375121/231552091-5ce6646a-ec77-42d6-8886-94237d621965.gif)

**Comment by HarelM** [comment]
The bounty.
